tests/test_io_utils.py

```diff
@@ -14,35 +15,67 @@
     "filename, mean, shape",
     [
         ("img.tif", 13577.494856, (2048, 2048, 3)),
-        ("img.jpg", 52.412425, (2048, 2048, 3)),
         ("img.png", 52.33736, (2048, 2048, 3)),
         ("img-8bit.jp2", 47.09060, (343, 343, 3)),
         ("img-15bit.jp2", 0.3041897, (1830, 1830)),
         ("img-16bit.jp2", 0.3041897, (1830, 1830)),
     ],
 )
 def test_img_read(input_folder: str, output_folder: str, filename: str, mean: float, shape: Tuple[int, ...]) -> None:
-    is_jpeg = filename.endswith("jpg")
     img = read_data(os.path.join(input_folder, filename))
 
+    if filename.endswith("txt"):
+        img = np.fromstring(img)
+
     assert img.shape == shape
+    assert np.mean(img) == approx(mean, abs=1e-4)
+    assert img.flags["WRITEABLE"], "Obtained numpy array is not writeable"
+
+    output_file_path = os.path.join(output_folder, filename)
+    write_data(output_file_path, img)
+    new_img = read_data(output_file_path)
+
+    assert np.array_equal(img, new_img), "Original and saved image are not the same"
 
-    if is_jpeg or python_implementation() != "PyPy":
-        assert np.mean(img) == approx(mean, abs=1e-4)
 
+@pytest.mark.parametrize("filename, mean, shape", [("img.jpg", 52.412425, (2048, 2048, 3))])
+def test_img_read_jpeg(
+    input_folder: str, output_folder: str, filename: str, mean: float, shape: Tuple[int, ...]
+) -> None:
+    img = read_data(os.path.join(input_folder, filename))
+
+    assert img.shape == shape
+    assert np.mean(img) == approx(mean, abs=1e-4)
     assert img.flags["WRITEABLE"], "Obtained numpy array is not writeable"
 
-    file_path = os.path.join(output_folder, filename)
+    output_file_path = os.path.join(output_folder, filename)
+
+    with pytest.warns(SHUserWarning):
+        write_data(output_file_path, img)
+
+
+def test_read_write_text() -> None:
+    with TempFS() as filesystem:
+        file_path = os.path.join(filesystem.root_path, "test-string.txt")
+
+        text = "sentinelhub-py is often shortened to sh-py"
+        write_data(file_path, text)
+        assert text == read_data(file_path)
+
+
+def test_read_write_xml() -> None:
+    with TempFS() as filesystem:
+        file_path = os.path.join(filesystem.root_path, "test-xml.xml")
```

Comment:
I believe `filesystem.getsyspath` would be nicer here.

Reply:
done